### ISSUE TITLE
Messaging queue

### ISSUE THREAD

**Issue by yanivbh1** [PostHog/posthog#14385]
First of all, we are using and enjoying PostHog. It's really a helpful product!

## In what situation are you experiencing subpar performance?
Events ingestion and export using 3rd party integrations

*Please describe.*
Recently I noticed a few performance issues as our usage grew, and after digging up a bit in the open-source repo, I found that you are using RabbitMQ and Celery. As PostHog continues to grow rapidly, I assume the need for more scale, multi-tenancy, and priority queues has risen. Therefor, I thought thats the perfect time to bring up [Memphis.dev](https://memphis.dev), which can help you get there in no-time, and with minor dev efforts, resources, and adaptations.

Minor note, we are also working on integrating Memphis within PostHog Apps.

## How to reproduce

No specific action has made.

## Environment

- [X] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: _please provide_

**Comment by weyert** [comment]
I thought it was using Kafka?

**Comment by yanivbh1** [comment]
> I thought it was using Kafka?

PostHog?

**Comment by neilkakkar** [comment]
Yep, we use Kafka for our ingestion queue; the celery queue is for internal tasks (like cohort recalculation).

Mind elaborating a bit more about what kind of perf issues you are facing?
